### PR TITLE
Feat/form

### PR DESCRIPTION
**Pull request by aesteves60**
Add component Form:
Attributes: initialValues
Methods: reset, submit, getFormErrors, getFormValues, isFormValid, setFieldError, getFieldError, setFieldValue, getFieldValue
Events: odsOnSubmit, odsOnReset

- Fix generate doc for allow Record type
- Fix button not found form in shadow dom
- Add parsing string to Record

### CODE REVIEW

File: packages/components/src/input/src/components/osds-input/core/controller.ts [hunk 1]
```diff
@@ -64,6 +68,10 @@ class OdsInputController {
   }
 
   private validateValue(value?: number): void {
+    if (this.component.type !== ODS_INPUT_TYPE.number) {
```
Comment (aesteves60) [inline packages/components/src/input/src/components/osds-input/core/controller.ts:71]: Fix warning `[OsdsInput] The value attribute must be a correct number`

File: packages/components/src/input/src/components/osds-input/osds-input.tsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ export class OsdsInput implements OdsInputAttribute, OdsInputEvent, OdsInputMeth
 
   @Prop({ reflect: true }) required?: boolean = DEFAULT_ATTRIBUTE.required;
 
-  @Prop({ reflect: true }) size?: ODS_COMMON_FIELD_SIZE = DEFAULT_ATTRIBUTE.size;
+  @Prop({ reflect: true }) size: ODS_COMMON_FIELD_SIZE = DEFAULT_ATTRIBUTE.size;
```
Comment (dpellier) [inline packages/components/src/input/src/components/osds-input/osds-input.tsx:86]: not required as in main PR

File: packages/components/src/password/src/components/osds-password/osds-password.tsx [hunk 3]
```diff
@@ -54,7 +54,7 @@ export class OsdsPassword implements OdsPasswordAttribute, OdsPasswordMethod  {
 
   @Prop({ reflect: true }) required?: boolean = DEFAULT_ATTRIBUTE.required;
 
-  @Prop({ reflect: true }) size?: ODS_COMMON_FIELD_SIZE = DEFAULT_ATTRIBUTE.size;
+  @Prop({ reflect: true }) size: ODS_COMMON_FIELD_SIZE = DEFAULT_ATTRIBUTE.size;
```
Comment (dpellier) [inline packages/components/src/password/src/components/osds-password/osds-password.tsx:57]: not required as in main PR

File: packages/components/src/input/src/components/osds-input/core/controller.ts [hunk 4]
```diff
@@ -54,6 +55,9 @@ class OdsInputController {
   onValueChange(value: OdsInputValue, oldValue?: OdsInputValue): void {
     this.logger.debug(`[input=${this.component.value}]`, 'value changed', { oldValue, value });
     this.assertValue(value);
+    if (this.component.inputEl) {
+      this.component.inputEl.value = value?.toString() ?? '';
+    }
     this.component.internals.setFormValue(value?.toString() ?? '');
```
Comment (dpellier) [inline packages/components/src/input/src/components/osds-input/core/controller.ts:61]: as in main PR, carefull with string cast (and on above line)

**Review by dpellier — APPROVED** [review]

File: packages/components/src/input/src/components/osds-input/core/controller.ts (hunk 4 above)
Comment (aesteves60) [reply packages/components/src/input/src/components/osds-input/core/controller.ts:61]: This is "normal" because in a native input, the value is a string

**Review by dpellier — CHANGES_REQUESTED** [review]
Ideally, before merge:
- remove .npmignore
- remove LICENCE file
- remove tsconfig.prod.json
- remove generate:licence task
- update `tsconfig` in typedoc.json
- copy new .gitignore from any other component

**Review by dpellier — APPROVED** [review]

**Pull request merged**